copter/source/docs/systemid-mode-ctrl-optimization.rst

```diff
@@ -0,0 +1,257 @@
+| Gain crossover frequency f_gc (Hz)     | 0.55      |  **1.84** | 1.76     | Larger is better  |
++----------------------------------------+-----------+-----------+----------+-------------------+
+| Tracking Rise Time (s)                 | 0.47      | 0.214     | **0.21** | Smaller is better |
++----------------------------------------+-----------+-----------+----------+-------------------+
+| Tracking Overshoot (%)                 | 17.16     | **0.00**  | 3.96     | Smaller is better |
++----------------------------------------+-----------+-----------+----------+-------------------+
+| Tracking Settling Time (s)             | 3.25      | **1.42**  | 1.65     | Smaller is better |
++----------------------------------------+-----------+-----------+----------+-------------------+
+
+The following two tables show the optimized flight controller parameters for the roll and yaw axis compared to the default parameters of ArduCopter.
+The defaults are also used as initial parameters for the optimization.
+As a reference, parameters obtained from an autotune flight with :ref:`AUTOTUNE_AGGR<AUTOTUNE_AGGR>` = 0.1 are also shown.
+It shows that the optimized parameters differ greatly from the default and autotune parameters.
+Again remember that the presented results are **only valid for the examined quadcopter** with a weight of 1.5 kg, an arm length of 22 cm and 9 inch propellers.
+The resulting parameters are specifically applicable for this copter configuration and **must not** be used for any other copter.
+
+Roll:
+
++-------------------------------------------+-----------------------+----------------------+------------------------+
+| Parameter                                 | Default               | Optimized            | Autotune               |
++-------------------------------------------+-----------------------+----------------------+------------------------+
+| :ref:`ATC_RAT_RLL_P<ATC_RAT_RLL_P>`       | 0.1350                | 0.123                | 0.240025               |
++-------------------------------------------+-----------------------+----------------------+------------------------+
+| :ref:`ATC_RAT_RLL_I<ATC_RAT_RLL_I>`       | 0.1350                | 1.0                  | 0.240025               |
++-------------------------------------------+-----------------------+----------------------+------------------------+
+| :ref:`ATC_RAT_RLL_D<ATC_RAT_RLL_D>`       | 0.0036                | 0.0053               | 0.007948               |
++-------------------------------------------+-----------------------+----------------------+------------------------+
+| :ref:`ATC_RAT_RLL_FF<ATC_RAT_RLL_FF>`     | 0                     | 0                    | 0                      |
++-------------------------------------------+-----------------------+----------------------+------------------------+
+| :ref:`ATC_RAT_RLL_FLTT<ATC_RAT_RLL_FLTT>` | 23.0000               | 50.256               | 5.0                    |
```

Comment:
How did FLTT get changed in autotune.  As far as I can tell from the code, this is not a filter that is changed during multicopter autotune.  Is this a typo?

Reply:
It's not changed in autotune, it is switched off during twitches however

Reply:
I don't believe this is true. Autotune is run in the linear region of the attitude controller. It has step size limits that attempt to ensure this is the case. The step test does not apply any acceleration limits or other input shaping / control model. In operation the acceleration limits are there to ensure the aircraft is safe when it is recovering from large angle errors. In normal operation and mauvers the acceleration limit in the attitude controller should not be reached.

It looks like you had to reduce the FLTT to 5 Hz to get the autotune results to be stable. That seems very strange to me.

It is possible that your system ID settings may have pushed the aircraft into an acceleration limited region and that has resulted in an inaccurate model.  However, if the model is based on the rate output -> gyro measurement then any controller limits or command model should not impact it. Maybe if the outputs are saturating, that might mess up the system ID result.

The key point is Autotune is designed to optimise the aircraft in the linear region, just as would expect your system ID model to represent. If your model is not stable with the autotuned results, with the same filter settings, then the model is not representative of the aircraft.